block/produce.go

```diff
@@ -165,18 +168,18 @@ func (m *Manager) produceBlock(allowEmpty bool, nextProposerInfo *nextProposerIn
 		nextProposerAddr     = m.State.Sequencers.Proposer.SettlementAddress
 		lastProposerBlock    = false // Indicates that the block is the last for the current seq. True during the rotation.
 	)
-	// if nextProposerInfo is set, we create a last block
-	if nextProposerInfo != nil {
+	// if nextProposer is set, we create a last block
+	if nextProposer != nil {
 		maxBlockDataSize = 0
-		proposerHashForBlock = nextProposerInfo.nextProposerHash
-		nextProposerAddr = nextProposerInfo.nextProposerAddr
+		proposerHashForBlock = nextProposer.Hash
+		nextProposerAddr = nextProposer.Address
```

Comment:
NextProposerAddr is confusing cause in dymint world it refers to the dymint pubkeyAddr but we're actually refering to the App Addr here.